.pre-commit-config.yaml

```diff
@@ -59,3 +59,69 @@ repos:
     - '--width=120'
     - '--preserve-quotes'
     types: [yaml]
+
+# Python: for all the helpers in scripts/*
+- repo: https://github.com/psf/black
+  rev: '22.1.0'
+  hooks:
+  - id: black
+    name: Python-format-black
+    args:
+    - '--line-length=79'
+- repo: https://github.com/pycqa/isort
+  rev: '5.10.1'
+  hooks:
+  # Available args: https://pycqa.github.io/isort/docs/configuration/options.html
+  - id: isort
+    name: Python-format-isort
+    args:
+    - '--profile=black'
+    - '--line-length=79'
+- repo: https://github.com/pycqa/flake8
+  rev: '4.0.1'
+  hooks:
+  - id: flake8
+    name: Python-lint-flake8
+    args:
+    - '--max-line-length=79'
+    - '--extend-ignore=E203' # ignore "whitespace before ':'"
+- repo: https://github.com/jendrikseipp/vulture
+  rev: 'v2.3'
+  hooks:
+  - id: vulture
+    name: Python-lint-vulture
+    types: [python]
+    pass_filenames: true
+    # excluding sphinx-conf as it would break to change it
+    exclude: docs/autogen/conf.py$
```

Comment:
I think we need a bit on config to make it actually trigger. But it doesn't break anything to have it in.

Reply:
Passing filenames will presumably make `vulture` only check for dead-code within each file. You need to supply directories/packages to check inter-dependability.